core/controller/src/main/scala/whisk/core/entitlement/Entitlement.scala

```diff
@@ -195,21 +200,45 @@ protected[core] abstract class EntitlementProvider(config: WhiskConfig, loadBala
   protected[core] def check(user: Identity, right: Privilege, resources: Set[Resource])(
     implicit transid: TransactionId): Future[Unit] = {
     val subject = user.subject
+    val inaccessibleResources = ListBuffer[Resource]()
 
     val entitlementCheck: Future[Boolean] = if (user.rights.contains(right)) {
       if (resources.nonEmpty) {
-        logging.info(this, s"checking user '$subject' has privilege '$right' for '${resources.mkString(",")}'")
+        logging.info(this, s"checking user '$subject' has privilege '$right' for '${resources.mkString(", ")}'")
         checkSystemOverload(right)
           .flatMap(_ => checkUserThrottle(user, right, resources))
           .flatMap(_ => checkConcurrentUserThrottle(user, right, resources))
           .flatMap(_ => checkPrivilege(user, right, resources))
+          .flatMap(resourcePrivSet => {
+            Future
+              .sequence {
+                resourcePrivSet.map { resourcePriv =>
```

Comment:
you could filter the resource priv set on `._2` and check if the resultant list is empty, this would save the map+forall later and the construction of `inaccessibleResources` explicitly.

Reply:
i wasn't sure how to escape the Future flatmap cycle to produce a non-Future map.  i made an update close to what you've suggested

Reply:
OK I will attempt this and can submit a patch.